core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedGlobalAggregationWithProjection.java

```diff
@@ -173,18 +178,25 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
                 return Result.empty();
             }
         }
-
         source = decorrelatedSource.get().getNode();
+        Symbol nonNull;
 
-        // append non-null symbol on nested plan. It will be used to restore semantics of null-sensitive aggregations after LEFT join
-        Symbol nonNull = context.getSymbolAllocator().newSymbol("non_null", BOOLEAN);
-        source = new ProjectNode(
-                context.getIdAllocator().getNextId(),
-                source,
-                Assignments.builder()
-                        .putIdentities(source.getOutputSymbols())
-                        .put(nonNull, TRUE)
-                        .build());
+        AggregationNode globalAggregation = captures.get(AGGREGATION);
+        Optional<Symbol> booleanAggregationSymbol = getBooleanAggregationSymbol(globalAggregation);
+        if (booleanAggregationSymbol.isEmpty()) {
+            // append non-null symbol on nested plan. It will be used to restore semantics of null-sensitive aggregations after LEFT join
+            nonNull = context.getSymbolAllocator().newSymbol("non_null", BOOLEAN);
+            source = new ProjectNode(
+                    context.getIdAllocator().getNextId(),
+                    source,
+                    Assignments.builder()
+                            .putIdentities(source.getOutputSymbols())
+                            .put(nonNull, TRUE)
+                            .build());
+        }
+        else {
```

Comment:
you most likely need separate, special execution path when `getBooleanAggregationSymbol`, because a lot of code below is unrelated and make is more tricky to support with `PushFilterThough..` (e.g. adding masks)